cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/LocalArtifactLoaderStage.java

```diff
@@ -113,8 +113,11 @@ public void process(AppDeploymentInfo deploymentInfo) throws Exception {
         metadataValidator.validateTags(appEntity, metadata.getTags());
       }
     }
+    ApplicationSpecification appSpecInStore = ApplicationId.DEFAULT_VERSION.equals(applicationId.getVersion())
```

Comment:
Not quite sure if the version is correctly passed in here...

Reply:
I believe the right thing to do here is always call store.getLatest()